doc/patch-author-guide.md

```diff
@@ -261,58 +224,76 @@ A shadow PID variable is allocated in `do_fork()`: it is associated with the
 current `struct task_struct *p` value, given a string lookup key of "newpid",
 sized accordingly, and allocated as per `GFP_KERNEL` flag rules.
 
-`kpatch_shadow_alloc` returns a pointer to the shadow variable, so we can
+`klp_shadow_get_or_alloc()` returns a pointer to the shadow variable, so we can
```

Comment:
FWIW, I had grabbed this from a local test/integration/fedora-30/shadow-newpid.patch, not realizing that it was never pushed up to the repository.  I think either call should work.  klp_shadow_get_or_alloc() is probably more straightforward.

Reply:
How would klp_shadow_get_or_alloc() be more straightforward here?  In do_fork() there shouldn't be an existing shadow struct for it, right?

Reply:
D'oh.. copy/paste error.  I meant to say klp_shadow_alloc() for that reason.

Reply:
Actually.... this bugged me as I found the same klp_shadow_get_or_alloc() usage in the rhel-7.x integration tests.  It traces back to the centos-7 tests in which @sm00th modified the test in 56be80720aee ("integration/centos-7: fix shadow-newpid.patch warnings").  From the commit message:
> When quickly loading/unloading this patch multiple times it is possible to hit "Duplicate shadow variable" warnings since the patch doesn't have any cleanup hooks on unload.
Switch to klp_shadow_get_or_alloc to ignore these.

I can provide the original version of the patch in the author guide using klp_shadow_alloc() and mention that it was adjusted for testing purposes or some other disclaimer.

Reply:
Ah, that makes sense now.  I guess the correct fix would be to use klp_shadow_free_all() in a post-unload callback?  If so it might be worth adding that to the real integration patches as well for proper test coverage.

Reply:
I would say that is the correct solution for a real kpatch, however I usually try hard not to mix features when writing test cases.  In this case, I think we could call klp_shadow_free_all() from a module_exit function to achieve the same result.